File: MP7/TablePartD.py
import happybase as hb

# DON'T CHANGE THE PRINT FORMAT, WHICH IS THE OUTPUT
# OR YOU WON'T RECEIVE POINTS FROM THE GRADER

connection = hb.Connection()
connection.open()

table = connection.table('powers')

# Id: "row1", Values for (hero, power, name, xp, color)
q1 = table.row(b'row1')

hero = q1[b'personal:hero']
power = q1[b'personal:power']
name = q1[b'professional:name']
xp = q1[b'professional:xp']
color = q1[b'custom:color']

print('hero: {}, power: {}, name: {}, xp: {}, color: {}'.format(hero, power, name, xp, color))

# Id: "row19", Values for (hero, color)
q19 = table.row(b'row19')

hero = q19[b'personal:hero']
color = q19[b'custom:color']

print('hero: {}, color: {}'.format(hero, color))

# Id: "row1", Values for (hero, name, color)

hero = q1[b'personal:hero']
name = q1[b'professional:name']
color = q1[b'custom:color']
print('hero: {}, name: {}, color: {}'.format(hero, name, color))

File: MP7V2_HBase/JavaTemplate/TablePartD.java
import java.io.IOException;

import org.apache.hadoop.conf.Configuration;

import org.apache.hadoop.hbase.HBaseConfiguration;
import org.apache.hadoop.hbase.HColumnDescriptor;
import org.apache.hadoop.hbase.HTableDescriptor;

import org.apache.hadoop.hbase.TableName;

import org.apache.hadoop.hbase.client.Admin;
import org.apache.hadoop.hbase.client.Connection;
import org.apache.hadoop.hbase.client.ConnectionFactory;
import org.apache.hadoop.hbase.client.Get;
import org.apache.hadoop.hbase.client.Put;
import org.apache.hadoop.hbase.client.Result;
import org.apache.hadoop.hbase.client.ResultScanner;
import org.apache.hadoop.hbase.client.Scan;
import org.apache.hadoop.hbase.client.Table;
import org.apache.hadoop.hbase.client.TableDescriptor;


import org.apache.hadoop.hbase.util.Bytes;

public class TablePartD{

   public static void main(String[] args) throws IOException {

	// TODO      
	// DON' CHANGE THE 'System.out.println(xxx)' OUTPUT PART
	// OR YOU WON'T RECEIVE POINTS FROM THE GRADER 
	
	String hero = ???;
	String power = ???;
	String name = ???;
	String xp = ???;
	String color = ???;
	System.out.println("hero: "+hero+", power: "+power+", name: "+name+", xp: "+xp+", color: "+color);

	hero = ???;
	color = ???;
	System.out.println("hero: "+hero+", color: "+color);

	hero = ???;
	name = ???;
	color = ???;
	System.out.println("hero: "+hero+", name: "+name+", color: "+color); 
   }
}

